nextest-runner/src/config/core/imp.rs

```diff
@@ -1124,6 +1150,71 @@ pub(in crate::config) struct NextestConfigImpl {
 }
 
 impl NextestConfigImpl {
+    /// Resolves a profile with inheritance chain
+    fn resolve_profile_chain(&self, profile_name: &str) -> Result<Vec<&CustomProfileImpl>, ProfileNotFound> {
+        let mut visited = HashSet::new();
+        let mut chain = Vec::new();
+        
+        self.resolve_profile_chain_recursive(profile_name, &mut visited, &mut chain)?;
+        Ok(chain)
+    }
+    
+    fn resolve_profile_chain_recursive<'cfg>(
+        &'cfg self,
+        profile_name: &str,
+        visited: &mut HashSet<String>,
+        chain: &mut Vec<&'cfg CustomProfileImpl>,
+    ) -> Result<(), ProfileNotFound> {
+        if visited.contains(profile_name) {
+            return Err(ProfileNotFound::new(
+                profile_name,
+                self.all_profiles().collect::<Vec<_>>(),
+            ));
+        }
+        
+        visited.insert(profile_name.to_string());
```

Comment:
Do you need the `to_string` here? Or can you store references to the profile name?

You might run into [variance](https://lifetime-variance.sunshowers.io/) issues with storing a reference, but I think you can have another lifetime parameter for that.